### ISSUE TITLE
redis backplane not working as expected

### ISSUE THREAD

**Issue by bigerock** [MichaCo/CacheManager#201]
i've set up a sample project to illustrate. i'm trying to force an item out of cache, update it manually in redis, and the re-get the item. first, that doesn't seem to be working at all. the item only seems to be retrieved if the app is restarted or the item falls out of cache. second, the backplane obviously isn't working either if i can't even get the item to be removed from cache on client 1.

if you could please have look it would be greatly appreciated.

https://github.com/bigerock/cacheManTest

**Comment by MichaCo** [comment]
Hi @bigerock, I don't get what you are trying to do. Your website project seems to work, although, it is not really easy to figure out what you are trying to achieve.

Could you at least list the steps to reproduce?

A more simplified (console) app would also be much easier to understand and to reproduce something ;)

**Comment by bigerock** [comment]
my apologies. at the core i am trying to test out the pub/sub between two clients. but in doing so i found out that it doesn't seem to be invalidating the runtime cache no matter what i do manually. i am unable to make a change in a cache object in redis, and then clear the cache object at the client and retrieve the new object from redis. 
do you know what i'm missing?

i can try and make another project, but i think it's demonstrable in the existing one if i'm not mistaken. 
thanks!

**Comment by MichaCo** [comment]
Yeah, I meant more a list of actions I should perform to reproduce your problem with that app.
Like, run 2 instances of this, click here, do this, open redis client, edit entry manually? 
I just don't know what you do with that app, for me, it just has many service endpoints and a ui with tons of links and all that seems to work ;)

**Comment by bigerock** [comment]
ok. 
1) run the app. (after putting in your redis params in web.config)
2) go to redis and update one of the values created, say _cache.test.GetString.50 for example
3) click on 'remove' on the app for the number of the value you just updated in redis (50 in the example)
4) click on 'get' for the same number
5) note that you will not get the updated value but the same one from before (50 in this example)

**Comment by MichaCo** [comment]
Ok, well, when you call .remove, the key gets removed from in-memory AND Redis. The next time you do a get in your application, you create a new key which again has value "50".

To your initial problem, assuming you expect that the backplane would invalidate the local cache when you do updates manually in Redis, no, that's not supported. 
That being said, the Redis cache in CacheManager can be configured to listen to `expire/evict` and `del` events. And, if configured properly so that keyspace notifications are enabled, keys will get removed from local cache in those cases.

In short, use CacheManager to update values in Redis, and the backplane will work. Otherwise, handle it manually yourself if you have to manually update keys, like listen to keyspace notifications
